From 9c8ca53f8134922c10d92028d5e0b419b613c5eb Mon Sep 17 00:00:00 2001
From: Don Dutile <ddutile@redhat.com>
Date: Thu, 9 Nov 2017 06:47:59 -0500
Subject: [PATCH 2/3] [netdrv] qed: Correct TM ILT lines in presence of VFs

Message-id: <20171109065423.8574-85-ddutile@redhat.com>
Patchwork-id: 197066
O-Subject: [PATCH RHEL-7.5 v2 084/468] qed: Correct TM ILT lines in presence of VFs
Bugzilla: 1462433 1499362
RH-Acked-by: Jerry Snitselaar <jsnitsel@redhat.com>
RH-Acked-by: Jarod Wilson <jarod@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

commit 70566b42e94ad7de1328272c4b8454320f794a10
Author: Mintz, Yuval <Yuval.Mintz@cavium.com>
Date:   Mon Apr 3 12:21:11 2017 +0300

    qed: Correct TM ILT lines in presence of VFs

    As of today there's no protocol supported that requires
    support from the TM hardware block and enables SRIOV,
    but we should still correct the calculation to reflect
    the lines required for such future VFs instead of changing
    the PF's own lines.

    Signed-off-by: Yuval Mintz <Yuval.Mintz@cavium.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1462433

Signed-off-by: Donald Dutile <ddutile@redhat.com>
Signed-off-by: Rafael Aquini <aquini@redhat.com>
---
 drivers/net/ethernet/qlogic/qed/qed_cxt.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: src/drivers/net/ethernet/qlogic/qed/qed_cxt.c
===================================================================
--- src.orig/drivers/net/ethernet/qlogic/qed/qed_cxt.c	2017-12-04 17:13:52.518256604 +0100
+++ src/drivers/net/ethernet/qlogic/qed/qed_cxt.c	2017-12-04 17:13:58.826170322 +0100
@@ -783,8 +783,8 @@
 
 		qed_ilt_cli_adv_line(p_hwfn, p_cli, p_blk, &curr_line,
 				     ILT_CLI_TM);
-		p_cli->pf_total_lines = curr_line - p_blk->start_line;
 
+		p_cli->vf_total_lines = curr_line - p_blk->start_line;
 		for (i = 1; i < p_mngr->vf_count; i++)
 			qed_ilt_cli_adv_line(p_hwfn, p_cli, p_blk, &curr_line,
 					     ILT_CLI_TM);
